grading/load_config_json.cpp

```diff
@@ -1276,13 +1292,15 @@ void AddSubmissionLimitTestCase(nlohmann::json &config_json) {
   bool has_limit_test = false;
 
   // count total points and search for submission limit testcase
-  nlohmann::json::iterator tc = config_json.find("testcases");
-  assert (tc != config_json.end());
-  for (unsigned int i = 0; i < tc->size(); i++) {
+  nlohmann::json::iterator testcases = config_json.find("testcases");
+  assert (testcases != config_json.end());
+  for (nlohmann::json::iterator tc = testcases->begin(); tc != testcases->end(); tc++) {
     //This input to testcase is only necessary if the testcase needs to retrieve its 'commands'
     std::string container_name = "";
-    TestCase my_testcase(config_json,i,container_name);
-    int points = (*tc)[i].value("points",0);
+    std::string testcase_id = tc->value("testcase_id", "");
+    assert(testcase_id != "");
+    TestCase my_testcase(*tc, testcase_id, container_name);
+    int points = tc->value("points", 0);
```

Comment:
Is this counting up the points on the itempools too?
It should be.
But not double counting testcases in the global array + itempool testcases.  Right?